project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/errors/SnapshotError.kt

```diff
@@ -0,0 +1,7 @@
+package org.plan.research.minimization.plugin.errors
+
+sealed interface SnapshotError<out T> {
+    data class Aborted<T>(val reason: T) : SnapshotError<T>
+    data class TransactionFailed(val error: Throwable) : SnapshotError<Nothing>
+    data class TransactionCreationFailed(val reason: String) : SnapshotError<Nothing>
```

Comment:
I'm suggesting using `enum(String)` for it. Since, if we'll need to parse a reason in future it could be useful